src/Shared/Controls/Storage/FolderBrowser.xaml

```diff
@@ -30,20 +30,21 @@
                 ToolTipService.ToolTip="Navigate up"
                 Visibility="{x:Bind GoToParentCommand.IsRunning, Mode=OneWay}" />
 
-            <ProgressRing IsActive="{x:Bind GoToParentCommand.IsRunning, Mode=OneWay}" Visibility="{x:Bind GoToParentCommand.IsRunning, Mode=OneWay}" />
-
             <TextBlock Text="{x:Bind Name, Mode=OneWay}" />
+
+            <ProgressRing IsActive="{x:Bind FetchingItems, Mode=OneWay}" />
```

Comment:
The XAML in this file was using commands and `x:Bind`-ing to data directly to avoid creating UI-specific properties like these. Why was this change needed?

Reply:
I can't properly remember but there was major bug with existing implementation where there was no busy indicator for a long running task while fetching folders
![image](https://github.com/Arlodotexe/strix-music/assets/15306909/6b9424a9-15f1-459e-b800-90b3d462fb90)

here's the commit of that time:
https://github.com/Arlodotexe/strix-music/pull/278/commits/501c9651dbe681e941b7116ece35540488c13432

Reply:
It's built into AsyncRelayCommand

Reply:
Instead of `IsFolderEmpty`, use the existing `IsZeroToBoolean` function to bind `CurrentFolderItems.Count`. The fixup for `FetchingItems` is a bit more complex, I'll do it later.

Reply:
Ah neat, removed IsFolderEmpty boolean and used InverseToVisibilityConverter on FetchingItems boolean, since we need to track the state of the "FetchingItems" we can't directly check CurrentFolderItems count which will be zero on control load as the content hasn't been fetched by then since it takes sometime. So the correct state of the folder Items count can only be determined after getting response from the ipfsClient.